### PR TITLE
JAXB fix to make plugin work on Java 11

### PR DESCRIPTION
**Pull request by jglick**
Subsumes #89.

### CODE REVIEW

**Comment by jglick** [comment]
```sh
mvn -Dtest=JCloudsArtifactManagerTest,JCloudsVirtualFileTest surefire:test
```

(with a live AWS account) works using both JDK 8 and JDK 11. Now to check actual behavior in the Jenkins class loader.

**Comment by jglick** [comment]
Nope, I checked `1.4-rc281.0ed1add4ffcd` in an actual installation on Java 11 and it does not work.

**Comment by jglick** [comment]
I think I get it. [This text](https://github.com/jenkinsci/jaxb-plugin/blob/master/README.adoc)

> It is currently not recommended to depend on it from your plugin. It is designed to be automatically added as a dependency when running Jenkins on a Java 11 runtime

is potentially misleading, because the entry in `split-plugins.txt`

```
jaxb 2.163 2.3.0 11
```

means that this happens only when the plugin was compiled against a pre-2.163 baseline—now meaning _any obsolete LTS_. If you use a current LTS, then you _do_ need to depend on it explicitly.

File: Jenkinsfile [hunk 1]
```diff
@@ -1,9 +1,8 @@
 if (infra.isRunningOnJenkinsInfra()) {
     // ci.jenkins.io
     buildPlugin(configurations: [
-    [ platform: "linux", jdk: "8", jenkins: null ],
-    [ platform: "linux", jdk: "8",  jenkins: "2.164.1", javaLevel: "8" ],
-    [ platform: "linux", jdk: "11", jenkins: "2.164.1", javaLevel: "8" ],
+        [platform: 'linux', jdk: '8'],
+        [platform: 'linux', jdk: '11'],
```
Comment (batmat) [inline Jenkinsfile]: :+1:

File: pom.xml [hunk 2]
```diff
@@ -228,6 +228,11 @@
       <version>0.10-alpha</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+        <groupId>io.jenkins.plugins</groupId>
+        <artifactId>jaxb</artifactId>
+        <version>2.3.0.1</version>
```
Comment (batmat) [inline pom.xml]: Agreed. The docs are indeed misleading.

When I wrote them 1) I had more in mind the end users that still should never have to install this plugin manually and 2) that no plugin would bump that quickly to that baseline, so didn't really care explaining  that indeed at some point plugins should generallly declare explicit dependencies on some detached plugins when they *actually* depend on them.

Thanks a lot for triggering this rather sooner than later Jesse. Better have someone like you hit this first than someone less versed into Jenkins' understandings.

**Pull request merged**